src/adapter/src/catalog/consistency.rs

```diff
@@ -348,6 +358,143 @@ impl CatalogState {
             Err(dependency_inconsistencies)
         }
     }
+
+    /// # Invariants
+    ///
+    /// * Every schema that exists in the `schemas_by_name` map, also exists in `schemas_by_id`.
+    /// * The name present in the `schemas_by_name` map matches the name in the associated `Schema`
+    ///   struct.
+    /// * All items that exist in a `Schema` struct, also exist in the `entries_by_id` map.
+    /// * Parsing the `create_sql` string from an `Entry` succeeds.
+    /// * The result of parsing the `create_sql` must return a single `Statement`.
+    /// * The schema name in the returned `Statement`, must match that of the `Schema` struct.
+    /// * The item from the parsed `create_sql` must be fully qualified.
+    ///
+    fn check_items(&self) -> Result<(), Vec<ItemInconsistency>> {
+        let mut item_inconsistencies = vec![];
+
+        for (db_id, db) in &self.database_by_id {
+            for (schema_name, schema_id) in &db.schemas_by_name {
+                // Make sure the schema themselves are consistent.
+                let Some(schema) = db.schemas_by_id.get(schema_id) else {
+                    item_inconsistencies.push(ItemInconsistency::MissingSchema {
+                        db_id: *db_id,
+                        schema_name: schema_name.clone(),
+                    });
+                    continue;
+                };
+                if schema_name != &schema.name.schema {
+                    item_inconsistencies.push(ItemInconsistency::KeyedName {
```

Comment:
uncovered

Reply:
Part of the consistency checker, pulled out of this PR